### PR TITLE
Bh gpu

### PR DESCRIPTION
**Pull request by fweik**
Fixes #

Description of changes:
 - 

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: src/core/actor/DipolarBarnesHut.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include "EspressoSystemInterface.hpp"
+#include "forces.hpp"
+#include "energy.hpp"
+
+#ifdef DIPOLAR_BARNES_HUT
+
+void activate_dipolar_barnes_hut(float epssq, float itolsq)
+{
+    if (dipolarBarnesHut)
+        free(dipolarBarnesHut);
```
Comment (hmenke) [inline src/core/actor/DipolarBarnesHut.cpp:34]: Mixing `new`/`delete` with `malloc`/`free` will most likely lead to a segfault.

File: src/core/actor/DipolarBarnesHut.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if (dipolarBarnesHut)
+    {
+        forceActors.remove(dipolarBarnesHut);
+        energyActors.remove(dipolarBarnesHut);
+        delete(dipolarBarnesHut);
+        dipolarBarnesHut=NULL;
+    }
+}
+
+DipolarBarnesHut *dipolarBarnesHut=0;
```
Comment (hmenke) [inline src/core/actor/DipolarBarnesHut.cpp:56]: `nullptr`

File: src/core/actor/DipolarBarnesHut.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#ifdef DIPOLAR_BARNES_HUT
+
+void activate_dipolar_barnes_hut(float epssq, float itolsq)
+{
+    if (dipolarBarnesHut)
+        free(dipolarBarnesHut);
+
+    // also necessary on 1 CPU or GPU, does more than just broadcasting
+    mpi_bcast_coulomb_params();
+    dipolarBarnesHut =new DipolarBarnesHut(espressoSystemInterface, epssq, itolsq);
```
Comment (hmenke) [inline src/core/actor/DipolarBarnesHut.cpp:38]: Why is there a global (!) pointer (!) to the method?  This design looks very broken.

**Review by hmenke — CHANGES_REQUESTED** [review]

**Comment by fweik** [comment]
@hmenke please comment on #1273, whis was just to trigger the CI.

**Comment by bogdan-tanygin** [comment]
Thanks, I agree with @hmenke comments and will introduce required changes within #1273 

Please also note, that I've originally just cloned the existing actor architecture: https://github.com/espressomd/espresso/blob/python/src/core/actor/DipolarDirectSum.cpp
Hence, it may require same improvements. I'll open the separate issue/PR after the BH one completion to merge same improvements unless someone wish to work on the DDS actor.

**Pull request closed without merging**